apps/meteor/app/push/server/push.ts

```diff
@@ -124,7 +199,28 @@ class PushClass {
 			// Send to GCM
 			// We do support multiple here - so we should construct an array
 			// and send it bulk - Investigate limit count of id's
-			if (this.options.gcm?.apiKey) {
+			// TODO: Remove this after the legacy provider is removed
+			const useLegacyProvider = settings.get<boolean>('Push_UseLegacy');
+
+			if (!useLegacyProvider) {
+				// override this.options.gcm.apiKey with the oauth2 token
+				const token = await this.getFcmAuthorizationToken();
+				const sendGCMOptions = {
+					...this.options,
+					gcm: {
+						...this.options.gcm,
+						apiKey: token,
+					},
+				};
+
+				sendFCM({
+					userTokens: app.token.gcm,
+					notification,
+					_replaceToken: this.replaceToken,
+					_removeToken: this.removeToken,
+					options: sendGCMOptions as RequiredField<PushOptions, 'gcm'>,
+				});
```

Comment:
You are sending an object, could the project id not be sent from here at the same time without breaking the other signature (like inside `options`)?

I ask because the setting (Push_google_api_credentials) is loaded in both places, at one place being validated, another, not (not an issue exactly since flow has the second time covered), both times deserialized while doesn't have to just for the project_id.